class-vip-support-api.php

```diff
@@ -0,0 +1,101 @@
+<?php
+
+class VIP_Support_REST_Controller {
+
+	private static $namespace;
+
+	function __construct() {
+		self::$namespace = 'vip/v1';
+		add_action( 'rest_api_init', [ $this, 'rest_api_init' ] );
+	}
+
+	function permission_callback() {
+		return wpcom_vip_go_rest_api_request_allowed( self::$namespace );
```

Comment:
Probably becomes moot if we move this library into mu-plugins but should we use a filter here? Or maybe add a `function_exists` check?

Reply:
We don't have a way to check if it's an a8c IP, only if the request is proxied. I'm not sure if that will be too restrictive, as a lot of internal tools will make requests from a8c IPs, but not via proxies (async jobs or the Go API, for example).